os/board/rtl8721csm/src/component/common/bluetooth/realtek/sdk/board/common/src/hci_adapter.c

```diff
@@ -34,7 +34,7 @@ extern bool hci_board_complete(void);
 /* ================================internal=========================== */
 bool hci_rtk_tx_cb(void)
 {
-    if (p_hci_rtk->tx_buf != NULL)
+    if (p_hci_rtk->tx_buf != NULL && up_interruptcontext() == false)

```

Comment:
Is it ok not to free allocated memory? I think this makes memory leak.

Reply:
I think you need to free it like `sched_free( )`.

Reply:
It does not mean that you can call sched_free here. I mean you can refer the way.

Reply:
Or Realtek should re-implement this logic not to call free in interrupt handler.